src/reactA11yInputElementsRule.ts

```diff
@@ -61,6 +61,21 @@ function isExcludedInputType(node: ts.JsxSelfClosingElement, attributes: { [prop
     return false;
 }
 
+function isInputTypeFile(node: ts.JsxSelfClosingElement, attributes: { [propName: string]: ts.JsxAttribute }): boolean {
```

Comment:
This method is almost the same as `isExcludedInputType`; the only difference is in the innermost `if` statement. You should be able to deduplicate the code a bit by passing in a `condition: (attributeTex: string) => boolean` to the method, or something similar. 

_(if you can think of a more elegant way, that'd be great!)_

Reply:
I didn't come up with anything 😁 but agree with your idea.